code/modules/economy/quikpay.dm

```diff
@@ -0,0 +1,206 @@
+	R.info += "<li>Have them swipe their card to pay for the items, press return to reset the sum.</li></ol>"
+
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
+	if(!R.stamped)
+		R.stamped = new
+	R.offset_x += 0
+	R.offset_y += 0
+	R.ico += "paper_stamp-cent"
+	R.stamped += /obj/item/weapon/stamp
+	R.add_overlay(stampoverlay)
+	R.stamps += "<HR><i>This paper has been stamped by the Head of Personnel's desk.</i>"
+
+/obj/item/device/nanoquikpay/AltClick(var/mob/user)
+	if(editmode == 1)
+		editmode = 0
+		to_chat(user, span("notice", "Device Locked."))
+		SSvueui.check_uis_for_change(src)
+		return 0
+	if(editmode == 0)
+		var/attempt_code = input("Enter the edit code", "Confirm edit access code") as num //Copied the eftpos method, dont judge
+		if(attempt_code == access_code)
+			editmode = 1
+			to_chat(user, span("notice", "Device Unlocked."))
+			SSvueui.check_uis_for_change(src)
+
+
+/obj/item/device/nanoquikpay/proc/print_reference()
+	var/obj/item/weapon/paper/R = new(src.loc)
+	var/pname = "Reference: [machine_id]"
+	var/info = "<b>[machine_id] reference</b><br><br>"
+	info += "Access code: [access_code]<br><br>"
+	info += "<b>Do not lose or misplace this code.</b><br>"
+	R.set_content_unsafe(pname, info)
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
+	if(!R.stamped)
+		R.stamped = new
+	R.stamped += /obj/item/weapon/stamp
+	R.add_overlay(stampoverlay)
+	R.stamps += "<HR><i>This paper has been stamped by the Head of Personnel's desk.</i>"
+	var/obj/item/smallDelivery/D = new(R.loc)
+	R.forceMove(D)
+	D.wrapped = R
+	D.name = "small parcel - 'Quik Pay access code'"
+
+
+/obj/item/device/nanoquikpay/proc/print_receipt()
+	var/obj/item/weapon/paper/R = new(usr.loc)
+	var/receiptname = "Receipt: [machine_id]"
+	R.set_content_unsafe(receiptname, receipt, sum)
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
+	if(!R.stamped)
+		R.stamped = new
+	R.stamped += /obj/item/weapon/stamp
+	R.add_overlay(stampoverlay)
+	R.stamps += "<HR><i>This paper has been stamped by the Quik-Pay device.</i>"
+
+
+
+/obj/item/device/nanoquikpay/attackby(obj/O, mob/user)
+	var/obj/item/weapon/card/id/I = O.GetID()
+	if (!I) 
+		return
+	if (!istype(O))
+		return
+
+	var/datum/money_account/quickpay_account = SSeconomy.get_department_account("Civilian")
+	var/datum/money_account/customer_account = SSeconomy.get_account(I.associated_account_number)
+	if (!customer_account)
+		to_chat(user, span("notice", "Unable to access account, please contact the Head of Personnel."))
+		return 0
+
+	if(customer_account.suspended)
+		to_chat(user, span("notice", "Account Suspended, please contact the Head of Personnel."))
+		return 0
+
+	if(customer_account.security_level != 0) //If card requires pin authentication (ie seclevel 1 or 2)
```

Comment:
Pin Authentification is only required on security level 2.
Security level 1 is pin or card